build/AzurePipelinesTemplates/MUX-RunHelixTests-Job.yml

```diff
@@ -67,6 +74,74 @@ jobs:
       filePath: build\Helix\PrepareHelixPayload.ps1
       arguments: -Platform '$(buildPlatform)' -Configuration '$(buildConfiguration)'
 
+  - task: powershell@2
+    displayName: 'Create RunTestsInHelix-InteractionTests.proj'
+    condition:
+      ne('${{parameters.testSuite}}','NugetTestSuite')
+    inputs:
+      targetType: filePath
+      filePath: build\Helix\GenerateTestProjFile.ps1
+      arguments: -TestFile '$(artifactsDir)\drop\$(buildConfiguration)\$(buildPlatform)\Test\MUXControls.Test.dll' -OutputProjFile '$(Build.SourcesDirectory)\build\Helix\RunTestsInHelix-InteractionTests.proj' -JobTestSuiteName 'DevTestSuite' -TaefPath '$(taefPath)'
+
+  - task: powershell@2
+    displayName: 'Create RunTestsInHelix-ApiTests.proj'
+    condition:
+      ne('${{parameters.testSuite}}','NugetTestSuite')
+    inputs:
+      targetType: filePath
+      filePath: build\Helix\GenerateTestProjFile.ps1
+      arguments: -TestFile '$(artifactsDir)\drop\$(buildConfiguration)\$(buildPlatform)\AppxPackages\MUXControlsTestApp_Test\MUXControlsTestApp.appx' -OutputProjFile '$(Build.SourcesDirectory)\build\Helix\RunTestsInHelix-ApiTests.proj' -JobTestSuiteName 'DevTestSuite' -TaefPath '$(taefPath)'
+
+  - task: powershell@2
+    displayName: 'Create RunTestsInHelix-IXMPTestAppTests.proj'
+    condition:
+      ne('${{parameters.testSuite}}','NugetTestSuite')
+    inputs:
+      targetType: filePath
+      filePath: build\Helix\GenerateTestProjFile.ps1
+      arguments: -TestFile '$(artifactsDir)\drop\$(buildConfiguration)\$(buildPlatform)\AppxPackages\IXMPTestApp_Test\IXMPTestApp.appx' -OutputProjFile '$(Build.SourcesDirectory)\build\Helix\RunTestsInHelix-IXMPTestAppTests.proj' -JobTestSuiteName 'DevTestSuite' -TaefPath '$(taefPath)'
+
+  - task: powershell@2
+    displayName: 'Create RunTestsInHelix-NugetTests.proj'
+    condition:
+      eq('${{parameters.testSuite}}','NugetTestSuite')
+    inputs:
+      targetType: filePath
+      filePath: build\Helix\GenerateTestProjFile.ps1
+      arguments: -TestFile '$(artifactsDir)\drop\$(buildConfiguration)\$(buildPlatform)\Test\MUXControls.ReleaseTest.dll' -OutputProjFile '$(Build.SourcesDirectory)\build\Helix\RunTestsInHelix-NugetTests.proj' -JobTestSuiteName 'NugetTestSuite' -TaefPath '$(taefPath)'
+
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish RunTestsInHelix-InteractionTests.proj'
+    condition:
+      ne('${{parameters.testSuite}}','NugetTestSuite')
+    inputs:
+      PathtoPublish: $(Build.SourcesDirectory)\build\Helix\RunTestsInHelix-InteractionTests.proj
+      artifactName: 'drop'
+
```

Comment:
Writing outputs into the source tree is something we like to avoid. Can we write into the artifactstagingdirectory, then publish that as one step? And teach the 'Run tests in helix' job to read the files from the other dir?

Reply:
Sure.  Just for my info for the future, why specifically do we want to avoid that?  Just to make sure we know what we're dealing with, or is there a more specific technical reason?

Reply:
No specific technical reason -- mostly just cleanliness. Also it ensures that you don't have to "git reset --hard" or anything like that once you've finished a build. You just have to nuke your BuildOutput folder to get back to clean.

---
In reply to: [280267753](https://github.com/Microsoft/microsoft-ui-xaml/pull/642#discussion_r280267753) [](ancestors = 280267753)